docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -619,6 +619,97 @@ You can also add any number of `else` blocks:
 {/if}
 ----
 
+[[when_section]]
+==== When/Switch Section
+
+This section is similar to Java's `switch` or Kotlin's `when` constructs.
+It matches a _tested value_ against all blocks sequentially until a condition is satisfied.
+The first matching block is executed.
+All other blocks are ignored (this behavior differs to the Java `switch` where a `break` statement is necessary).
+
+.Example using the `when`/`is` name aliases
+[source]
+----
+{#when items.size}
+  {#is 1} <1>
+    There is exactly one item!
+  {#is > 10} <2>
+    There are more than 10 items!
+  {#else} <3>
+    There are 2 -10 items!
+{/when}
+----
+<1> If there is exactly one parameter it's tested for equality.
+<2> It's possible to use <<when_operators,an operator>> to specify the matching logic. Unlike in the <<if_section>> nested operators are not supported.
+<3> `else` is block is executed if no other block matches the value.
+
+.Example using the `switch`/`case` name aliases
+[source]
+----
+{#switch person.name}
+  {#case 'John'} <1>
+    Hey John!
+  {#case 'Mary'} <1>
+    Hey Mary!
+{/switch}
+----
+<1> `case` is an alias for `is`.
+
+A tested value that resolves to an enum is handled specifically.
+The parameters of an `is`/`case` block are not evaluated as expressions but compared with the result of `toString()` invocation upon the tested value.
```

Comment:
I guess that works.

Reply:
But if you make this generic and say that "if a comparison operator operand (including implicit operands such as in `case/is`) is an enum then both operands are compared using `toString` on both operands" then this works for every operator in every expression context.

Reply:
Well, currently there is no such thing like "expression context". 

We have "expressions" that are used to output a single value, i.e. `{foo}`. These expressions basically follow the `object.property` (dot notation) or `object[property_name]` (bracket notation) syntax, plus the property can be replaced with a virtual method, e.g. `{foo.bar(1)}` and we also support "infix notation" for virtual methods with single parameter, i.e. `{name or 'John'}`. 

And then we have sections, e.g. `{#if}` which can have any number of parameters (with optional names) and it's completely up to the section to decide what to do with these params. For example `{#for item in foo.items}` has 3 params, `item`, `in` and `foo.items`... first is used as an alias, second is ignored and third is used to resolve an iterable object...